lib/Doctrine/Migrations/Configuration/Loader/XmlFileLoader.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Migrations\Configuration\Loader;
+
+use Doctrine\Migrations\Configuration\Configuration;
+use Doctrine\Migrations\Configuration\Exception\FileNotFound;
+use Doctrine\Migrations\Configuration\Exception\XmlNotValid;
+use Doctrine\Migrations\Tools\BooleanStringFormatter;
+use DOMDocument;
+use SimpleXMLElement;
+use const DIRECTORY_SEPARATOR;
+use const LIBXML_NOCDATA;
+use function assert;
+use function file_exists;
+use function file_get_contents;
+use function libxml_use_internal_errors;
+use function simplexml_load_string;
+use function strtr;
+
+class XmlFileLoader extends AbstractFileLoader
+{
+    /** @var ArrayLoader */
+    private $arrayLoader;
+
+    /**
+     * @return mixed[]
+     */
+    private function extractParameters(SimpleXMLElement $root, bool $nodes = true) : array
+    {
+        $config = [];
+        foreach ($nodes ? $root->children() : $root->attributes() as $node) {
+            $nodeName = strtr($node->getName(), '-', '_');
+            if ($nodeName === 'migrations_paths') {
+                $config['migrations_paths'] = [];
+                foreach ($node->{'path'} as $pathNode) {
+                    $config['migrations_paths'][(string) $pathNode['namespace']] = (string) $pathNode;
+                }
+            } elseif ($nodeName === 'storage' && $node->{'table-storage'} instanceof SimpleXMLElement) {
+                $config['table_storage'] = $this->extractParameters($node->{'table-storage'}, false);
+            } else {
+                $config[$nodeName] = (string) $node;
+            }
+        }
+
+        return $config;
+    }
+
+    /**
+     * @param mixed $file
+     */
+    public function __construct(?ArrayLoader $arrayLoader = null)
+    {
+        $this->arrayLoader = $arrayLoader ?: new ArrayLoader();
+    }
+
+    /**
+     * @param mixed|string $file
+     */
+    public function load($file) : Configuration
+    {
+        if (! file_exists($file)) {
+            throw FileNotFound::new();
+        }
+        libxml_use_internal_errors(true);
```

Comment:
the existing state must be restored (using a `try/finally`) to avoid changing the global state.

Reply:
btw, I think 2.x suffers from this too.